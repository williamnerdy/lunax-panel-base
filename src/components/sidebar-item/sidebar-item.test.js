import React from 'react';
import { shallow } from 'enzyme';
import SidebarItem from './sidebar-item';

describe('<SidebarItem />', () => {
  let wrapper, link, icon;

  beforeEach(() => {
    wrapper = shallow(
      <SidebarItem url="/test" icon="test">
        Test
      </SidebarItem>
    );
    link = wrapper.find('Link');
    icon = wrapper.find('Icon');
  });

  it('shoud render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('shoud have "Test" on text', () => {
    expect(wrapper.contains('Test')).toBe(true);
  });

  it('shoud have a link', () => {
    expect(link.length).toBe(1);
  });

  it('shoud have "/test" on url', () => {
    expect(link.props().to).toBe('/test');
  });

  it('shoud have a icon', () => {
    expect(icon.length).toBe(1);
  });

  it('shoud have "test" on icon', () => {
    expect(icon.props().type).toBe('test');
  });
});
