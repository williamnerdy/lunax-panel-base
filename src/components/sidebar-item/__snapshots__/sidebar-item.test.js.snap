// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SidebarItem /> shoud render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SidebarItem
    icon="test"
    url="/test"
  >
    Test
  </SidebarItem>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Link
        replace={false}
        to="/test"
      >
        <Icon
          type="test"
        />
        <span
          className="nav-text"
        >
          Test
        </span>
      </Link>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Icon
            type="test"
          />,
          <span
            className="nav-text"
          >
            Test
          </span>,
        ],
        "replace": false,
        "to": "/test",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "type": "test",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Test",
            "className": "nav-text",
          },
          "ref": null,
          "rendered": "Test",
          "type": "span",
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Link
          replace={false}
          to="/test"
        >
          <Icon
            type="test"
          />
          <span
            className="nav-text"
          >
            Test
          </span>
        </Link>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Icon
              type="test"
            />,
            <span
              className="nav-text"
            >
              Test
            </span>,
          ],
          "replace": false,
          "to": "/test",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "type": "test",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Test",
              "className": "nav-text",
            },
            "ref": null,
            "rendered": "Test",
            "type": "span",
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
